src/main/java/net/sf/rails/util/RunGame.java

```diff
@@ -1,38 +1,54 @@
 package net.sf.rails.util;
 
+import java.awt.*;
+import java.awt.desktop.OpenFilesEvent;
+import java.awt.desktop.OpenFilesHandler;
 import java.io.File;
 
+import javax.swing.*;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import net.sf.rails.common.ConfigManager;
+import net.sf.rails.ui.swing.AutoSaveLoadDialog;
 import net.sf.rails.ui.swing.GameSetupController;
 
 public class RunGame {
 
+    private static final Logger log = LoggerFactory.getLogger(RunGame.class);
+
     public static void main(String[] args) {
+        final String[] fileName = { null };
```

Comment:
Can we replace this with a `List<String>`? I'm not a friend of arrays containing `null` values

Reply:
This type of usage is common when you need to reference & change/set a variable within an anonymous class where it's a single item, ie String, Integer/int, etc. Creation of a List is confusing as it leads the developer/reader to think there may be multiple items. I'm also of the mind set that creation of a List (ie the object creation) to hold a single item is a code smell and is overhead not needed. In this case, I don't see there is much difference (ir it was possible) between setting a String to null and setting the first element in the array to null. 

Maybe it's just from my background of C/C++ but nulls in arrays aren't a big deal to me :)

Reply:
Ah I understand now. Yes lambda functions can only access effectively `final` variables.

> Maybe it's just from my background of C/C++ but nulls in arrays aren't a big deal to me :)

For me it is quite the opposite. I've done a lot with `scala` and there you try to not use `null` when possible. Instead you can use `Option` with its two flavors `Some` and `None` to symbolize whether a value is set or not.

In Java I think it is fine to make use of `null` as long as it does not looks too crazy :)

Reply:
Of course after this whole discussion I rewrote this logic anyway (for a different reason) in an upcoming PR ;)

Java has something similar, and in fact I think they stole the idea from Scala, `Optional`, although I personally don't see much value in it as even with it you still end up having to make an `if` check, it's just supposedly a bit prettier because you call `isPresent()`. Again, probably my C/C++ background but I think eyeballing logic and seeing that `== null` is clearer to me then my brain having to process the fact that a method is being called, and that method is basically making the `null` check when I see `isPresent()`. Maybe it makes more sense for developers that haven't had extensive experience with `null` in other languages but for me, blah.